BlackJack.py

```diff
@@ -0,0 +1,369 @@
+                    print("%s have: %s, %s points" % (player.name, player.hand, player.count_score()))
+                    choice = self.get_user_choice(player)
+                    player.last_move = MoveType.match_move(choice)
+                    self.game_action(choice, player)
+
+    def get_user_choice(self, player):
+        if player.cash >= player.bet_value and player.last_move == MoveType.NONE:
+            return (UserInputProvider().collect_proper_str_from_user(MoveType.values_no_split_up_low(),
+                                                                         "%s, please enter h for hit, st for stand or dd for double down: " % player.name)).lower()
+        else:
+            return (UserInputProvider().collect_proper_str_from_user(MoveType.values_no_split_double_up_low(),
+                                                                     "%s, please enter h for hit or st for stand: " % player.name)).lower()
+
+    def game_action(self, choice, player):
+        if choice == MoveType.HIT.value:
+            player.add_card_to_hand(self.deck.get_card())
+        elif choice == MoveType.DOUBLE.value:
+            player.add_card_to_hand(self.deck.get_card())
+            self.croupier.get_money_from_player(player)
+            player.set_last_move(MoveType.DOUBLE)
+            print("You have %s, %s points and your bet value now is equal: %s" % (
+                player.hand, player.count_score(), player.bet_value))
+
+    def get_croupier_cards(self):
+        while self.croupier.should_draw_cards():
+            print("Croupier have: %s, %s points" % (self.croupier.hand, self.croupier.count_score()))
+            self.croupier.add_card_to_hand(self.deck.get_card())
+        else:
+            print("Croupier have: %s, %s points" % (self.croupier.hand, self.croupier.count_score()))
+
+    def print_croupier_card(self):
+        print("Croupier have: ", self.croupier.get_single_card())
+
+    def players_betting(self):
+        for player in self.players:
+            print("%s have: %s, %s points" % (player.name, player.hand, player.count_score()))
+            player.set_bet_value(UserInputProvider().collect_int_in_range_min_max_from_user(1, player.cash,
+                                                                                      "%s please enter your bet bigger than 1$: " % player.name))
+            self.croupier.get_money_from_player(player)
+
+    def add_croupier_to_players(self):
+        return self.players.append(self.croupier)
+
+    def remove_players_from_all_players(self, players):
+        [self.players.remove(player) for player in players]
+
+    def get_busters(self):
+        self.add_croupier_to_players()
+        busters = [player for player in self.players if player.count_score() > 21]
+        self.remove_players_from_all_players(busters)
+        return busters
+
+    def get_scores(self, player):
+        return player.count_score()
+
+    def sort_players(self, players, boolean):
```

Comment:
Boolean, to nazwa typu, a nie zmiennej. Co to za zmienna? Czy players jest kiedykolwiek inne od self.players? Jeśli nie, to zbędny parametr

Reply:
Dodatkowo, jeśli korzystasz tylko z jednej opcji sortowania, to można ustawić defaultowa wartość tego parametru, żeby później przy wykorzystaniu nie trzeba było podawać